### ISSUE TITLE
Can not create timedelta64 object from datetime.timedelta(0) with a resolution argument

### ISSUE THREAD

**Issue by will133** [numpy/numpy#11624]
There seems to be a problem with the cast checking code where I would see this behavior (on numpy 1.14.2 and Python 3.6.6):

```
Python 3.6.6 |Anaconda, Inc.| (default, Jun 28 2018, 17:14:51)
[GCC 7.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy
>>> import datetime
>>> numpy.__version__
'1.14.2'

>>> numpy.timedelta64(datetime.timedelta(1), 'us')
numpy.timedelta64(86400000000,'us')

>>> numpy.timedelta64(datetime.timedelta(0), 'us')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Cannot cast datetime.timedelta object from metadata [Y] to [us] according to the rule 'same_kind'
```
I do not quite understand why this would not work.  I was actually using numpy 1.11.1 before the upgrade and the code seemed to work fine.

If you do not give the microsecond argument it would work:

```
>>> numpy.timedelta64(datetime.timedelta(0))
numpy.timedelta64(0,'us')
```

**Comment by will133** [comment]
BTW, I tried 1.11.3 and this problems starts to appear in that version (so probably can narrow down the changeset).

**Comment by charris** [comment]
Fixed in master, and probably the current 1.15.
```
In [1]: numpy.timedelta64(datetime.timedelta(0), 'us')
Out[1]: numpy.timedelta64(0,'us')

In [2]: np.__version__
Out[2]: '1.16.0.dev0+44cfca8'
```

**Comment by charris** [comment]
Thanks for reporting.
